tests/integration_rust/search_tests.rs

```diff
@@ -65,3 +66,64 @@ async fn search_return_latest_across_everything() {
 
     assert_eq!(found_package.package_record.version.as_str(), "4");
 }
+
+#[tokio::test]
+async fn test_search_multiple_versions() {
+    let mut package_database = PackageDatabase::default();
+
+    // Add package with multiple versions and build strings
+    package_database.add_package(
+        Package::build("foo", "0.1.0")
+            .with_build("h60d57d3_0")
+            .with_subdir(Platform::NoArch)
+            .finish(),
+    );
+    package_database.add_package(
+        Package::build("foo", "0.1.0")
+            .with_build("h60d57d3_1")
+            .with_subdir(Platform::NoArch)
+            .finish(),
+    );
+    package_database.add_package(
+        Package::build("foo", "0.2.0")
+            .with_build("h60d57d3_0")
+            .with_subdir(Platform::NoArch)
+            .finish(),
+    );
+    package_database.add_package(
+        Package::build("foo", "0.2.0")
+            .with_build("h60d57d3_1")
+            .with_subdir(Platform::NoArch)
+            .finish(),
+    );
+    let temp_dir = TempDir::new().unwrap();
+    let channel_dir = temp_dir.path().join("channel");
+    package_database.write_repodata(&channel_dir).await.unwrap();
+    let channel = Url::from_file_path(channel_dir).unwrap();
+    let platform = Platform::current();
+    let pixi = PixiControl::from_manifest(&format!(
+        r#"
+    [project]
+    name = "test-multiple-versions"
+    channels = ["{channel}"]
+    platforms = ["{platform}"]
+
+    "#
+    ))
+    .unwrap();
+
+    let mut out = Vec::new();
+    let builder = pixi.search("foo".to_string());
+    let result = search::execute_impl(builder.args, &mut out)
+        .await
+        .unwrap()
+        .unwrap();
+    let output = String::from_utf8(out).unwrap();
+
+    assert_eq!(result.len(), 1);
```

Comment:
do you think it makes sense to snapshot result via `insta`?

Reply:
I think there's no additional value provided by the snapshot. Why do you think it's helpful?

Reply:
then you would more easily see whether the whitespaces of our output were set correctly

Reply:
Ah wait, I thought you meant the line `assert_eq!(result.len(), 1);` which only checks for repo data records. We could move the assertions regarding `output` into a snapshot, yes.

Reply:
[76cb5d1](https://github.com/prefix-dev/pixi/pull/2440/commits/76cb5d1f28a8f88d50af5e493c28878240ed6cb2)